// catalog/lib/components/Link/HeadingLink.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Link = require('./Link');

var _Link2 = _interopRequireDefault(_Link);

var _CatalogPropTypes = require('../../CatalogPropTypes');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var style = function style(theme) {
  return {
    headingLink: {
      color: theme.lightColor,
      fill: theme.lightColor,
      ':hover': {
        color: theme.linkColor,
        fill: theme.linkColor,
        textDecoration: 'none'
      }
    }
  };
};

var HeadingLink = function HeadingLink(_ref, _ref2) {
  var catalog = _ref2.catalog;

  var slug = _ref.slug,
      rest = _objectWithoutProperties(_ref, ['slug']);

  return _react2.default.createElement(
    _Link2.default,
    _extends({ className: 'HeadingLink', title: 'Link to this section', to: '#' + slug, 'aria-hidden': true, style: style(catalog.theme).headingLink }, rest),
    '#'
  );
};

HeadingLink.propTypes = {
  slug: _react.PropTypes.string.isRequired
};

HeadingLink.contextTypes = {
  catalog: _CatalogPropTypes.catalogShape.isRequired
};

exports.default = HeadingLink;
