// catalog/lib/specimens/ReactSpecimen/ReactSpecimen.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CatalogPropTypes = require('../../CatalogPropTypes');

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _Frame = require('../../components/Frame/Frame');

var _Frame2 = _interopRequireDefault(_Frame);

var _Specimen = require('../../components/Specimen/Specimen');

var _Specimen2 = _interopRequireDefault(_Specimen);

var _HighlightedCode = require('../../components/HighlightedCode/HighlightedCode');

var _HighlightedCode2 = _interopRequireDefault(_HighlightedCode);

var _ResponsiveTabs = require('../../components/ResponsiveTabs/ResponsiveTabs');

var _ResponsiveTabs2 = _interopRequireDefault(_ResponsiveTabs);

var _Hint = require('../../specimens/Hint');

var _Hint2 = _interopRequireDefault(_Hint);

var _reactElementToString = require('./reactElementToString');

var _reactElementToString2 = _interopRequireDefault(_reactElementToString);

var _transformJSX = require('../../utils/transformJSX');

var _transformJSX2 = _interopRequireDefault(_transformJSX);

var _validateSizes = require('../../utils/validateSizes');

var _validateSizes2 = _interopRequireDefault(_validateSizes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

function getStyle(theme) {
  return {
    container: {
      border: '1px solid #eee',
      boxSizing: 'border-box',
      fontFamily: theme.fontFamily,
      margin: '0 0 20px 0',
      position: 'relative',
      width: '100%'
    },
    content: {
      background: 'url(' + theme.checkerboardPatternLight + ')',
      border: 'none',
      borderRadius: '2px',
      boxSizing: 'border-box',
      display: 'block',
      padding: 20,
      position: 'relative',
      width: '100%'
    },
    light: {
      background: 'url(' + theme.checkerboardPatternLight + ')'
    },
    dark: {
      background: 'url(' + theme.checkerboardPatternDark + ')'
    },
    plain: {
      background: 'transparent',
      padding: '0'
    },
    plain_light: {
      background: theme.bgLight,
      padding: '20px'
    },
    plain_dark: {
      background: theme.bgDark,
      padding: '20px'
    },
    responsive: {
      background: 'url(' + theme.checkerboardPatternLight + ')',
      boxSizing: 'border-box',
      overflow: 'hidden',
      padding: '15px',
      textAlign: 'center'
    }
  };
}

var ReactSpecimen = function (_Component) {
  _inherits(ReactSpecimen, _Component);

  function ReactSpecimen(props) {
    _classCallCheck(this, ReactSpecimen);

    var _this = _possibleConstructorReturn(this, _Component.call(this, props));

    _this.state = {
      elementState: props.state,
      parentWidth: null,
      activeScreenSize: (0, _validateSizes2.default)(props.responsive, props.catalog.responsiveSizes)[0] || null
    };
    _this.setElementState = _this.setElementState.bind(_this);
    _this.setSize = _this.setSize.bind(_this);
    _this.updateParentWidth = _this.updateParentWidth.bind(_this);
    return _this;
  }

  ReactSpecimen.prototype.componentDidMount = function componentDidMount() {
    if (this.state.activeScreenSize) {
      window.addEventListener('resize', this.updateParentWidth);
      setTimeout(this.updateParentWidth);
    }
  };

  ReactSpecimen.prototype.componentWillUnmount = function componentWillUnmount() {
    if (this.state.activeScreenSize) {
      window.removeEventListener('resize', this.updateParentWidth);
    }
  };

  ReactSpecimen.prototype.setElementState = function setElementState(nextState) {
    if (typeof nextState === 'function') {
      this.setState(function (_ref) {
        var elementState = _ref.elementState;
        return { elementState: _extends({}, elementState, nextState(elementState)) };
      });
    } else {
      this.setState({ elementState: _extends({}, this.state.elementState, nextState) });
    }
  };

  ReactSpecimen.prototype.updateParentWidth = function updateParentWidth() {
    var nextParentWidth = this.specimen.getBoundingClientRect().width - 30;
    if (nextParentWidth !== this.state.parentWidth) {
      this.setState({ parentWidth: nextParentWidth });
    }
  };

  ReactSpecimen.prototype.setSize = function setSize(activeScreenSize) {
    this.setState({ activeScreenSize: activeScreenSize });
  };

  ReactSpecimen.prototype.render = function render() {
    var _this2 = this;

    var _props = this.props,
        _props$catalog = _props.catalog,
        imports = _props$catalog.page.imports,
        theme = _props$catalog.theme,
        responsiveSizes = _props$catalog.responsiveSizes,
        children = _props.children,
        noSource = _props.noSource,
        frame = _props.frame,
        options = _objectWithoutProperties(_props, ['catalog', 'children', 'noSource', 'frame']);

    var _state = this.state,
        activeScreenSize = _state.activeScreenSize,
        parentWidth = _state.parentWidth;

    var styles = getStyle(theme);
    var validSizes = (0, _validateSizes2.default)(options.responsive, responsiveSizes);

    var exampleStyles = _extends({}, options.plain ? styles.plain : null, options.light ? styles.light : null, options.dark ? styles.dark : null, options.plain && options.light ? styles.plain_light : null, options.plain && options.dark ? styles.plain_dark : null, options.responsive ? styles.responsive : null);

    var jsx = typeof children === 'string';
    var element = null;
    var error = null;
    var code = '';

    if (jsx) {
      var transformed = (0, _transformJSX2.default)(children, _extends({}, imports, {
        state: this.state.elementState,
        setState: this.setElementState
      }));
      element = transformed.element;
      error = transformed.error ? _react2.default.createElement(
        _Hint2.default,
        { warning: true },
        'Couldn\'t render specimen: ' + transformed.error
      ) : null;
      code = children;
    } else {
      element = children;
      code = (0, _reactElementToString2.default)(children);
    }

    if (options.responsive && !validSizes) {
      return _react2.default.createElement(
        _Hint2.default,
        { warning: true },
        'Please check that the responsive parameters match an existing entry.'
      );
    }

    if (error) return error;

    return _react2.default.createElement(
      'section',
      { style: styles.container, ref: function ref(el) {
          _this2.specimen = el;
        } },
      options.responsive && parentWidth && activeScreenSize && _react2.default.createElement(_ResponsiveTabs2.default, { theme: theme, sizes: validSizes, action: this.setSize, activeSize: activeScreenSize, parentWidth: parentWidth }),
      (!options.responsive || parentWidth) && _react2.default.createElement(
        'div',
        { style: _extends({}, styles.content, exampleStyles) },
        frame || activeScreenSize ? _react2.default.createElement(
          _Frame2.default,
          { width: activeScreenSize && activeScreenSize.width, parentWidth: parentWidth ? parentWidth : '100%', height: activeScreenSize && activeScreenSize.height },
          element
        ) : element
      ),
      !noSource && _react2.default.createElement(_HighlightedCode2.default, { language: 'jsx', code: code, theme: theme })
    );
  };

  return ReactSpecimen;
}(_react.Component);

ReactSpecimen.propTypes = {
  catalog: _CatalogPropTypes.catalogShape.isRequired,
  children: _react.PropTypes.oneOfType([_react.PropTypes.element, _react.PropTypes.string]).isRequired,
  responsive: _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.string, _react.PropTypes.arrayOf(_react.PropTypes.string)]),
  noSource: _react.PropTypes.bool,
  plain: _react.PropTypes.bool,
  light: _react.PropTypes.bool,
  dark: _react.PropTypes.bool,
  frame: _react.PropTypes.bool,
  state: _react.PropTypes.object
};

exports.default = (0, _Specimen2.default)(undefined, undefined, { withChildren: true })((0, _radium2.default)(ReactSpecimen));
