// catalog/lib/components/ResponsiveTabs/ResponsiveTabs.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Preview = require('./Preview');

var _Preview2 = _interopRequireDefault(_Preview);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function getStyle(theme) {
  return {
    tabContainer: {
      alignItems: 'center',
      background: 'white',
      borderBottom: '1px solid #eee',
      display: 'flex',
      overflowX: 'auto',
      width: '100%'
    },
    tab: {
      alignItems: 'center',
      background: '#eee',
      boxSizing: 'border-box',
      color: '#777',
      cursor: 'pointer',
      display: 'flex',
      flexBasis: '100%',
      flexDirection: 'row',
      padding: '10px',
      transition: '.2s background-color, .4s color'
    },
    tabActive: {
      background: 'white',
      fontWeight: 'bold',
      color: theme.brandColor,
      cursor: 'auto'
    },
    description: {
      paddingLeft: 5
    },
    tabDimension: {
      color: '#777',
      display: 'block',
      fontFamily: theme.fontMono,
      fontSize: 'smaller',
      fontWeight: 'normal',
      marginTop: 2,
      opacity: 0.6
    }
  };
}

var ResponsiveTabs = function ResponsiveTabs(_ref) {
  var sizes = _ref.sizes,
      action = _ref.action,
      activeSize = _ref.activeSize,
      theme = _ref.theme,
      parentWidth = _ref.parentWidth;

  var styles = getStyle(theme);
  return _react2.default.createElement(
    'div',
    { style: styles.tabContainer },
    sizes.map(function (val, i) {
      var isTabActive = activeSize.name === val.name;
      var activeStyles = isTabActive && styles.tabActive;
      return _react2.default.createElement(
        'div',
        { key: i, style: _extends({}, styles.tab, activeStyles), onClick: function onClick() {
            return action(val);
          } },
        _react2.default.createElement(_Preview2.default, { proportion: val.width / val.height }),
        _react2.default.createElement(
          'div',
          { style: styles.description },
          val.name,
          _react2.default.createElement(
            'div',
            { style: styles.tabDimension },
            val.width,
            '\xD7',
            val.height,
            '\u2009',
            parentWidth <= val.width && '(scaled)'
          )
        )
      );
    })
  );
};

ResponsiveTabs.propTypes = {
  sizes: _react.PropTypes.array,
  action: _react.PropTypes.func,
  activeSize: _react.PropTypes.object,
  theme: _react.PropTypes.object,
  parentWidth: _react.PropTypes.number
};

exports.default = ResponsiveTabs;
