// catalog/lib/components/Page/PageHeader.js
'use strict';

exports.__esModule = true;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _radium = require('radium');

var _radium2 = _interopRequireDefault(_radium);

var _typography = require('../../styles/typography');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var PageHeader = function (_Component) {
  _inherits(PageHeader, _Component);

  function PageHeader() {
    _classCallCheck(this, PageHeader);

    return _possibleConstructorReturn(this, _Component.apply(this, arguments));
  }

  PageHeader.prototype.render = function render() {
    var _props = this.props,
        theme = _props.theme,
        title = _props.title,
        superTitle = _props.superTitle;


    var styles = {
      outerHeader: {
        boxSizing: 'border-box',
        position: 'relative',
        height: theme.pageHeadingHeight,
        background: theme.pageHeadingBackground
      },
      innerHeader: {
        position: 'absolute',
        bottom: theme.sizeL,
        left: theme.sizeL,
        '@media (min-width: 1000px)': {
          left: theme.sizeL * 2
        }
      },
      superTitle: _extends({}, (0, _typography.heading)(theme, 1), {
        color: theme.pageHeadingTextColor,
        opacity: 0.6,
        margin: 0
      }),
      title: _extends({}, (0, _typography.heading)(theme, 4), {
        color: theme.pageHeadingTextColor,
        margin: 0
      })
    };

    return _react2.default.createElement(
      'div',
      { style: styles.outerHeader },
      _react2.default.createElement(
        'div',
        { style: styles.innerHeader },
        _react2.default.createElement(
          'h2',
          { style: styles.superTitle },
          superTitle
        ),
        _react2.default.createElement(
          'h1',
          { style: styles.title },
          title
        )
      )
    );
  };

  return PageHeader;
}(_react.Component);

PageHeader.propTypes = {
  theme: _react.PropTypes.object.isRequired,
  title: _react.PropTypes.string.isRequired,
  superTitle: _react.PropTypes.string.isRequired
};

exports.default = (0, _radium2.default)(PageHeader);
