// catalog/lib/components/Page/PageRenderer.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CatalogPropTypes = require('../../CatalogPropTypes');

var _Page = require('./Page');

var _Page2 = _interopRequireDefault(_Page);

var _runscript = require('../../utils/runscript');

var _runscript2 = _interopRequireDefault(_runscript);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var renderStyles = function renderStyles(styles) {
  return styles.map(function (src, i) {
    return _react2.default.createElement('link', { key: i, href: src, rel: 'stylesheet', type: 'text/css' });
  });
};

var renderContent = function renderContent(content) {
  return _react2.default.isValidElement(content) && content.type === _Page2.default ? content : _react2.default.createElement(
    _Page2.default,
    null,
    content
  );
};

var PageRenderer = function (_Component) {
  _inherits(PageRenderer, _Component);

  function PageRenderer() {
    _classCallCheck(this, PageRenderer);

    var _this = _possibleConstructorReturn(this, _Component.call(this));

    _this.jump = _this.jump.bind(_this);
    return _this;
  }

  PageRenderer.prototype.componentDidMount = function componentDidMount() {
    this.context.catalog.page.scripts.forEach(_runscript2.default);
    this.jump();
  };

  PageRenderer.prototype.componentDidUpdate = function componentDidUpdate() {
    this.context.catalog.page.scripts.forEach(_runscript2.default);
    this.jump();
  };

  PageRenderer.prototype.jump = function jump() {
    var _this2 = this;

    var _props$location = this.props.location,
        a = _props$location.query.a,
        hash = _props$location.hash;

    var selector = hash || '#' + a;
    clearTimeout(this.jumpTimeout);

    // Don't freak out when hash is not a valid selector (e.g. #/foo)
    try {
      (function () {
        var el = document.querySelector(selector);
        if (el) {
          // Defer scrolling by one tick (when the page has completely rendered)
          _this2.jumpTimeout = setTimeout(function () {
            return el.scrollIntoView();
          }, 0);
        }
      })();
    } catch (e) {
      // eslint-disable-line no-empty
    }
  };

  PageRenderer.prototype.render = function render() {
    var content = this.props.content;
    var styles = this.context.catalog.page.styles;

    return _react2.default.createElement(
      'div',
      null,
      renderStyles(styles),
      renderContent(content)
    );
  };

  return PageRenderer;
}(_react.Component);

PageRenderer.propTypes = {
  content: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.element]).isRequired,
  location: _react.PropTypes.object.isRequired
};

PageRenderer.contextTypes = {
  catalog: _CatalogPropTypes.catalogShape.isRequired
};

exports.default = PageRenderer;
