// catalog/lib/specimens.js
'use strict';

exports.__esModule = true;

var _RawCode = require('./specimens/RawCode');

var _RawCode2 = _interopRequireDefault(_RawCode);

var _Audio = require('./specimens/Audio');

var _Audio2 = _interopRequireDefault(_Audio);

var _Code = require('./specimens/Code');

var _Code2 = _interopRequireDefault(_Code);

var _Color = require('./specimens/Color');

var _Color2 = _interopRequireDefault(_Color);

var _ColorPalette = require('./specimens/ColorPalette');

var _ColorPalette2 = _interopRequireDefault(_ColorPalette);

var _Html = require('./specimens/Html');

var _Html2 = _interopRequireDefault(_Html);

var _Hint = require('./specimens/Hint');

var _Hint2 = _interopRequireDefault(_Hint);

var _Image = require('./specimens/Image');

var _Image2 = _interopRequireDefault(_Image);

var _Table = require('./specimens/Table');

var _Table2 = _interopRequireDefault(_Table);

var _Type = require('./specimens/Type');

var _Type2 = _interopRequireDefault(_Type);

var _Download = require('./specimens/Download');

var _Download2 = _interopRequireDefault(_Download);

var _Video = require('./specimens/Video');

var _Video2 = _interopRequireDefault(_Video);

var _ReactSpecimen = require('./specimens/ReactSpecimen/ReactSpecimen');

var _ReactSpecimen2 = _interopRequireDefault(_ReactSpecimen);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  'raw-code': _RawCode2.default,
  audio: _Audio2.default,
  code: _Code2.default,
  color: _Color2.default,
  'color-palette': _ColorPalette2.default,
  html: _Html2.default,
  hint: _Hint2.default,
  image: _Image2.default,
  table: _Table2.default,
  type: _Type2.default,
  download: _Download2.default,
  video: _Video2.default,
  react: _ReactSpecimen2.default
};
